src/asl.ts

```diff
@@ -2067,31 +2250,39 @@ export class ASL {
             "Expected left side of assignment to be a variable."
           );
         }
-        return this.evalContext(expr, (evalExpr) => {
+        return this.evalContext(expr, ({ evalExpr }) => {
           const right = evalExpr(expr.right);
           const left = evalExpr(expr.left);
 
           if (!ASLGraph.isJsonPath(left)) {
             throw new SynthError(
               ErrorCodes.Unexpected_Error,
-              `Expected assignment to target a variable, found: ${left.value}`
+              `Expected assignment to target a variable, found: ${
+                ASLGraph.isLiteralValue(left) ? left.value : "boolean"
+              }`
             );
           }
 
-          return {
-            ...ASLGraph.passWithInput(
-              {
-                Type: "Pass",
-                ResultPath: left.jsonPath,
-                Next: ASLGraph.DeferNext,
-              },
-              right
-            ),
-            output: left,
-          };
+          return this.assignValue(expr, right, left.jsonPath);
+        });
+      } else if (expr.op === "in") {
+        return this.evalContext(expr, ({ evalExpr }) => {
+          const left = evalExpr(expr.left);
+          const right = evalExpr(expr.right);
+          const elm = this.assertElementAccessConstant(left);
```

Comment:
nit: Does this return a different value than `left`? Should it be an `asserts` function instead?

Reply:
it does return a different value, changed to `getElementAccessConstant`